### DISCUSSION TITLE
Event before tab is leaving

### DISCUSSION THREAD

**Discussion by tgeber** [quasarframework/quasar#15449] [General - Components / Directives / etc] [answered]
Hello everyone,

is it somehow possible to intercept an event before switching from one q-tab to another and to prevent the switch under certain circumstances? Something like @onBeforeChange or similar.

Can this event handling to an existing Quasar  component be added by myself? How does something like this work, if possible?

I absolutely need this functionality, which is why I would really appreciate your help. Thank you all.

Greetings from Germany sorry for my english, 
Thomas

I'm using Quasar v.2.11 with Vue 3 an Pinia. If relevant, I use the Composition API.

**Comment by tgeber** [comment]
Can nobody help?

**Comment by suleimanas** [comment]
you can always use (watch)

```
import { ref, watch } from "vue";
const tab = ref("tab1");

watch(tab, (newTab, oldTab)=>{
//
});
```

**Reply by tgeber** [reply]
The change of tab is then already carried out. But I have to be able to prevent this.

**Reply by suleimanas** [reply]
you have the old value you can change it back - i would do it like this:
1. i would watch the tab ref
2. on change if it pass then i would activate what it's shows panel or div or anything, otherwise it'll be inactive on every change 
3. if not pass just set the old value back or any defaults

**Reply by tgeber** [reply]
Unfortunately it is not possible to change back. There is a form in my tab and I want to allow leaving the tab when the data in the form has changed only with a message that needs to be confirmed.

**Comment by metalsadman** [comment, marked as answer]
do it in parent QTabs, use `:model-value` `@update:model-value="handler"` https://codepen.io/metalsadman/pen/qBMdezx

**Reply by tgeber** [reply]
Wow, that's exactly what I was looking for. Thank you very much.
